### DISCUSSION TITLE
Error when running msw in github-pages

### DISCUSSION THREAD

**Discussion by gaisdav** [mswjs/msw#1222] [Q&A] [answered]
Hi, I have error when running msw in my github-pages. I use create-react-app on my public repository.
link to gh-page: https://gaisdav.github.io/yul-yort-admin/#/login
```
Worker for scope ('https://gaisdav.github.io/') with script ('https://gaisdav.github.io/mockServiceWorker.js'): Service Worker script does not exist at the given path.

Did you forget to run "npx msw init <PUBLIC_DIR>"?
```
Screenshot of error:
<img width="574" alt="Screenshot 2022-04-29 at 00 08 07" src="https://user-images.githubusercontent.com/49130957/165827974-227e8898-b35f-4318-9a87-686ad6611ccf.png">

Code, how I use msw on src/index.tsx
<img width="629" alt="Screenshot 2022-04-29 at 00 09 39" src="https://user-images.githubusercontent.com/49130957/165828209-3eb7a681-6d38-4fe9-8bca-cb1368254f24.png">

Maybe someone can help me?

**Comment by kettanaito** [comment]
Hey, @gaisdav. 

As the error suggests, you need to serve the `mockServiceWorker.js` file from your website, even if it's a static GitHub page. You can find more info about that [on this page](https://mswjs.io/docs/getting-started/integrate/browser). Once you do, the client-side part of the library will be able to register the worker script and enable mocking.

It's also worth mentioning that you rarely need API mocking in production. While you may use it for educational purposes, or for the purpose of mocking API calls on an otherwise static site, it's generally discouraged to deploy MSW to production. Just be cautious about this.

**Reply by gaisdav** [reply]
Hey, [kettanaito](https://github.com/kettanaito).
Thanks for the reply, it seems that static files is in the yul-yort-admin directory. And msw looks for mockServiceWorker.js at the root.
Thank you for sharing the link, I will study further.

**Comment by gaisdav** [comment, marked as answer]
I found what was the problem.
It was necessary to write the correct path to the mockServiceWorker.js file. 
I have attached a link to the documentation below.

https://mswjs.io/docs/getting-started/integrate/browser#using-homepage-property-in-packagejson

**Reply by kettanaito** [reply]
I'm glad you've resolved it! 🎉
